library/Booster/JsonRpc.hs

```diff
@@ -245,14 +246,31 @@ respond stateVar =
             solver <- traverse (SMT.initSolver def) mSMTOptions
 
             result <- case internalised of
-                Left patternErrors -> do
+                Left [] -> do
+                    Log.logErrorNS
+                        "booster"
+                        "Error internalising cterm."
+                    Log.logOtherNS
+                        "booster"
+                        (Log.LevelOther "ErrorDetails")
+                        (prettyPattern req.state.term)
+                    pure $
+                        Left $
+                            RpcError.backendError $
+                                RpcError.CouldNotVerifyPattern $
+                                    RpcError.ErrorOnly "Error internalising cterm."
+                Left (patternError : _) -> do
                     Log.logErrorNS "booster" $
-                        "Error internalising cterm: " <> Text.pack (show patternErrors)
+                        "Error internalising cterm: " <> pack (show patternError)
                     Log.logOtherNS
                         "booster"
                         (Log.LevelOther "ErrorDetails")
                         (prettyPattern req.state.term)
-                    pure $ Left $ RpcError.backendError RpcError.CouldNotVerifyPattern patternErrors
+                    pure $
+                        Left $
+                            RpcError.backendError $
+                                RpcError.CouldNotVerifyPattern $
+                                    patternErrorToRpcError patternError
```

Comment:
These `patternErrors` come from a call to `internaliseTermOrPredicate` so there will be one for the `TermAndPredicates`  case and one for the `Predicates` case. The new code should return both errors if at all possible.

Reply:
we will have to modify the error type again. before we were just throwing these into the data field haphazadly